// guildedapi-types.ts/v1.mjs
import mod from "./v1.js";

export default mod;
export const APIChannelCategories = mod.APIChannelCategories;
export const APIGuildCategories = mod.APIGuildCategories;
export const CHANNEL = mod.CHANNEL;
export const CHANNELS = mod.CHANNELS;
export const CHANNEL_ANNOUNCEMENT = mod.CHANNEL_ANNOUNCEMENT;
export const CHANNEL_ANNOUNCEMENTS = mod.CHANNEL_ANNOUNCEMENTS;
export const CHANNEL_ANNOUNCEMENT_COMMENT = mod.CHANNEL_ANNOUNCEMENT_COMMENT;
export const CHANNEL_ANNOUNCEMENT_COMMENTS = mod.CHANNEL_ANNOUNCEMENT_COMMENTS;
export const CHANNEL_ANNOUNCEMENT_COMMENT_EMOTE = mod.CHANNEL_ANNOUNCEMENT_COMMENT_EMOTE;
export const CHANNEL_ANNOUNCEMENT_COMMENT_EMOTES = mod.CHANNEL_ANNOUNCEMENT_COMMENT_EMOTES;
export const CHANNEL_ANNOUNCEMENT_EMOTE = mod.CHANNEL_ANNOUNCEMENT_EMOTE;
export const CHANNEL_ANNOUNCEMENT_EMOTES = mod.CHANNEL_ANNOUNCEMENT_EMOTES;
export const CHANNEL_ARCHIVE = mod.CHANNEL_ARCHIVE;
export const CHANNEL_DOC = mod.CHANNEL_DOC;
export const CHANNEL_DOCS = mod.CHANNEL_DOCS;
export const CHANNEL_DOC_COMMENT = mod.CHANNEL_DOC_COMMENT;
export const CHANNEL_DOC_COMMENTS = mod.CHANNEL_DOC_COMMENTS;
export const CHANNEL_DOC_COMMENT_EMOTE = mod.CHANNEL_DOC_COMMENT_EMOTE;
export const CHANNEL_DOC_COMMENT_EMOTES = mod.CHANNEL_DOC_COMMENT_EMOTES;
export const CHANNEL_DOC_EMOTE = mod.CHANNEL_DOC_EMOTE;
export const CHANNEL_DOC_EMOTES = mod.CHANNEL_DOC_EMOTES;
export const CHANNEL_EVENT = mod.CHANNEL_EVENT;
export const CHANNEL_EVENTS = mod.CHANNEL_EVENTS;
export const CHANNEL_EVENT_COMMENT = mod.CHANNEL_EVENT_COMMENT;
export const CHANNEL_EVENT_COMMENTS = mod.CHANNEL_EVENT_COMMENTS;
export const CHANNEL_EVENT_COMMENT_EMOTE = mod.CHANNEL_EVENT_COMMENT_EMOTE;
export const CHANNEL_EVENT_COMMENT_EMOTES = mod.CHANNEL_EVENT_COMMENT_EMOTES;
export const CHANNEL_EVENT_EMOTE = mod.CHANNEL_EVENT_EMOTE;
export const CHANNEL_EVENT_EMOTES = mod.CHANNEL_EVENT_EMOTES;
export const CHANNEL_EVENT_EVENT_SERIES = mod.CHANNEL_EVENT_EVENT_SERIES;
export const CHANNEL_EVENT_EVENT_SERIES_ENTITY = mod.CHANNEL_EVENT_EVENT_SERIES_ENTITY;
export const CHANNEL_EVENT_RSVP = mod.CHANNEL_EVENT_RSVP;
export const CHANNEL_EVENT_RSVPS = mod.CHANNEL_EVENT_RSVPS;
export const CHANNEL_MESSAGE = mod.CHANNEL_MESSAGE;
export const CHANNEL_MESSAGES = mod.CHANNEL_MESSAGES;
export const CHANNEL_MESSAGE_EMOTE = mod.CHANNEL_MESSAGE_EMOTE;
export const CHANNEL_MESSAGE_EMOTES = mod.CHANNEL_MESSAGE_EMOTES;
export const CHANNEL_MESSAGE_PIN = mod.CHANNEL_MESSAGE_PIN;
export const FORUM_TOPIC = mod.FORUM_TOPIC;
export const FORUM_TOPICS = mod.FORUM_TOPICS;
export const FORUM_TOPIC_COMMENT = mod.FORUM_TOPIC_COMMENT;
export const FORUM_TOPIC_COMMENTS = mod.FORUM_TOPIC_COMMENTS;
export const FORUM_TOPIC_COMMENT_EMOTE = mod.FORUM_TOPIC_COMMENT_EMOTE;
export const FORUM_TOPIC_COMMENT_EMOTES = mod.FORUM_TOPIC_COMMENT_EMOTES;
export const FORUM_TOPIC_EMOTE = mod.FORUM_TOPIC_EMOTE;
export const FORUM_TOPIC_LOCK = mod.FORUM_TOPIC_LOCK;
export const FORUM_TOPIC_PIN = mod.FORUM_TOPIC_PIN;
export const GUILD = mod.GUILD;
export const GUILD_BAN = mod.GUILD_BAN;
export const GUILD_BANS = mod.GUILD_BANS;
export const GUILD_CATEGORY = mod.GUILD_CATEGORY;
export const GUILD_CATEGORY_CREATE = mod.GUILD_CATEGORY_CREATE;
export const GUILD_CATEGORY_ROLE_PERMISSION = mod.GUILD_CATEGORY_ROLE_PERMISSION;
export const GUILD_CATEGORY_ROLE_PERMISSIONS = mod.GUILD_CATEGORY_ROLE_PERMISSIONS;
export const GUILD_CATEGORY_USER_PERMISSION = mod.GUILD_CATEGORY_USER_PERMISSION;
export const GUILD_CATEGORY_USER_PERMISSIONS = mod.GUILD_CATEGORY_USER_PERMISSIONS;
export const GUILD_CHANNEL_ROLE_PERMISSION = mod.GUILD_CHANNEL_ROLE_PERMISSION;
export const GUILD_CHANNEL_ROLE_PERMISSIONS = mod.GUILD_CHANNEL_ROLE_PERMISSIONS;
export const GUILD_CHANNEL_USER_PERMISSION = mod.GUILD_CHANNEL_USER_PERMISSION;
export const GUILD_CHANNEL_USER_PERMISSIONS = mod.GUILD_CHANNEL_USER_PERMISSIONS;
export const GUILD_GROUP = mod.GUILD_GROUP;
export const GUILD_GROUPS = mod.GUILD_GROUPS;
export const GUILD_GROUP_ALTERNATIVE = mod.GUILD_GROUP_ALTERNATIVE;
export const GUILD_GROUP_MEMBER = mod.GUILD_GROUP_MEMBER;
export const GUILD_GROUP_MEMBERS = mod.GUILD_GROUP_MEMBERS;
export const GUILD_MEMBER = mod.GUILD_MEMBER;
export const GUILD_MEMBERS = mod.GUILD_MEMBERS;
export const GUILD_MEMBER_BULK_XP = mod.GUILD_MEMBER_BULK_XP;
export const GUILD_MEMBER_PERMISSION = mod.GUILD_MEMBER_PERMISSION;
export const GUILD_MEMBER_ROLE = mod.GUILD_MEMBER_ROLE;
export const GUILD_MEMBER_ROLES = mod.GUILD_MEMBER_ROLES;
export const GUILD_MEMBER_ROLE_XP = mod.GUILD_MEMBER_ROLE_XP;
export const GUILD_MEMBER_SOCIALS = mod.GUILD_MEMBER_SOCIALS;
export const GUILD_MEMBER_XP = mod.GUILD_MEMBER_XP;
export const GUILD_ROLE = mod.GUILD_ROLE;
export const GUILD_ROLES = mod.GUILD_ROLES;
export const GUILD_ROLE_UPDATE_PERMISSION = mod.GUILD_ROLE_UPDATE_PERMISSION;
export const GUILD_SUBSCRIPTION = mod.GUILD_SUBSCRIPTION;
export const GUILD_SUBSCRIPTIONS = mod.GUILD_SUBSCRIPTIONS;
export const GUILD_WEBHOOK = mod.GUILD_WEBHOOK;
export const GUILD_WEBHOOKS = mod.GUILD_WEBHOOKS;
export const GatewayEvents = mod.GatewayEvents;
export const GatewayOPCodes = mod.GatewayOPCodes;
export const GatewayURL = mod.GatewayURL;
export const GatewayVersion = mod.GatewayVersion;
export const LIST_ITEM = mod.LIST_ITEM;
export const LIST_ITEMS = mod.LIST_ITEMS;
export const LIST_ITEM_COMPLETE = mod.LIST_ITEM_COMPLETE;
export const MEMBER_NICKNAME = mod.MEMBER_NICKNAME;
export const Permissions = mod.Permissions;
export const URL_SIGNATURES = mod.URL_SIGNATURES;
export const USER = mod.USER;
export const USER_SERVERS = mod.USER_SERVERS;
export const USER_STATUS = mod.USER_STATUS;
export const UserCategories = mod.UserCategories;
